### ISSUE TITLE
Composer init readline support

### ISSUE THREAD

**Issue by clojens** [composer/composer#3534]
Any chance perhaps someone knows either a good 'readline'-like program (e.g. rlwrap or highline) that will function in PHP or has their finger on the readline deal in PHP?

Because the `composer init` is driving me insane on typos not being able to do simple backspace (history not even needed so much but would be nice). I know not *everyone* has that enabled but any self-respecting tool should (imho) allow full line editing capabilities.

In other words: can pretty please someone knowledgeable fix this part of the script some time, when they have time. Not priority but still warranted a issue I believe :)

Cheers

**Comment by theravel** [comment]
I think it can be easily done with Symfony Console Shell https://github.com/symfony/Console/blob/master/Shell.php
Just wrap Application into Shell and call "run" method.

Here is how it would look:
![screenshot from 2015-01-21 11 45 12](https://cloud.githubusercontent.com/assets/1029434/5833356/a356a912-a163-11e4-833a-8ab94cd734ea.png)

But this would totally change user experience for composer, I'm not sure we need this

**Comment by Seldaek** [comment]
Not sure I get wha tyou mean actually, with composer init I can backspace just fine when I make a typo.
